src/test/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtilsUnitTest.java

```diff
@@ -1905,6 +1912,41 @@ public void testOnTheFlyTabixCreation() throws IOException {
         }
     }
 
+    @Test
+    public void testCreateVcfWriterOnNio() throws IOException {
+        try (FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix())) {
+            final Path outputGZIP = jimfs.getPath("testOnTheFlyTabixCreation.vcf.gz");
+            final Path tabixIndex = outputGZIP.resolveSibling(outputGZIP.getFileName().toString() + TabixUtils.STANDARD_INDEX_EXTENSION);
+
+            final File inputGZIPFile = new File(
+                publicTestDir + "org/broadinstitute/hellbender/engine/8_mutect2_sorted.vcf.gz");
+
+            long recordCount = 0;
+            try (final VariantContextWriter vcfWriter = GATKVariantContextUtils.createVCFWriter(
+                outputGZIP,
+                null,
+                false,
+                Options.INDEX_ON_THE_FLY);
+                final FeatureReader<VariantContext> inputFileReader =
+                    AbstractFeatureReader
+                        .getFeatureReader(inputGZIPFile.getAbsolutePath(), new VCFCodec(), false)) {
+                vcfWriter.writeHeader((VCFHeader) inputFileReader.getHeader());
+                final Iterator<VariantContext> it = inputFileReader.iterator();
+                while (it.hasNext()) {
+                    vcfWriter.add(it.next());
+                    recordCount++;
+                }
+            }
+
+            // make sure we got an output and tabix index
+            Assert.assertTrue(Files.exists(outputGZIP));
+            Assert.assertTrue(Files.exists(tabixIndex));
+            Assert.assertTrue(Files.size(tabixIndex) > 0);
+
+            // We can't verify the index via query because it doesn't currently take Path :(
```

Comment:
You can use VCFFileReader(path) to query the result for the index test.

Reply:
Indeed, thank you! I missed it (was expecting all the public methods to be on top, should have looked more closely).